src/verifai/samplers/domain_sampler.py

```diff
@@ -52,8 +60,27 @@ def __init__(self, domain, samplers):
         self.samplers = tuple(samplers)
 
     def nextSample(self, feedback=None):
-        return self.domain.rejoinPoints(
```

Comment:
Do we really need this? You should be able to implement `getSample` and `update` and then get `nextSample` for free, as indicated above.